lib/shrine/plugins/add_metadata.rb

```diff
@@ -94,6 +94,7 @@ def _metadata_proc(name, &block)
       module InstanceMethods
         def extract_metadata(io, context)
           metadata = super
+          context = context.merge(metadata: metadata)
```

Comment:
Looks good, though I think you'd want to move it inside the block:

```rb
metadata = super

opts[:metadata].each do |metadata_block|
  custom_metadata = instance_exec(io, context.merge(metadata: metadata), &metadata_block) || {}
  # ...
end
```

Reply:
> so that each add_metadata block also has the metadata extracted from the previous add_metadata block.

So, they already do, and the tests actually prove it. 

This is because they all share the SAME (mutable) hash. When one `add_metadata` adds metadata, it's added to that very same "outside of block" hash, which is then passed in to the next. 

Is this a problem in any way, do you think?

Reply:
Oh, you’re totally right, I didn’t see it before. Yeah, that looks great then.